integration/query_set_test.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2021 FerretDB Set.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type testCase struct {
+	id     string
+	update bson.D
+	result bson.D
+	err    *mongo.WriteError
+	stat   *mongo.UpdateResult
+	alt    string
+}
+
+func TestSetOperatorOnString(t *testing.T) {
+	t.Parallel()
+
+	for name, tc := range map[string]testCase{
+		"Many": {
+			id:     "string",
+			update: bson.D{{"$set", bson.D{{"foo", int32(1)}, {"bar", bson.A{}}}}},
+			stat: &mongo.UpdateResult{
+				MatchedCount:  1,
+				ModifiedCount: 1,
+				UpsertedCount: 0,
+			},
+			result: bson.D{{"_id", "string"}, {"value", "foo"}, {"bar", bson.A{}}, {"foo", int32(1)}},
+		},
+		"NilOperand": {
+			id:     "string",
+			update: bson.D{{"$set", nil}},
+			err: &mongo.WriteError{
+				Code: 9,
+				Message: "Modifiers operate on fields but we found type null instead. " +
+					"For example: {$mod: {<field>: ...}} not {$set: null}",
+			},
+		},
+		"String": {
+			id:     "string",
+			update: bson.D{{"$set", "string"}},
+			err: &mongo.WriteError{
+				Code: 9,
+				Message: "Modifiers operate on fields but we found type string instead. " +
+					"For example: {$mod: {<field>: ...}} not {$set: \"string\"}",
+			},
+			alt: "Modifiers operate on fields but we found type string instead. " +
+				"For example: {$mod: {<field>: ...}} not {$set: string}",
+		},
+		"Double": {
+			id:     "string",
+			update: bson.D{{"$set", float64(42.12345)}},
```

Comment:
I think that this test, and several following it, does not add match compared to the string test above

Reply:
removed.